executorlib/task_scheduler/interactive/spawner_pysqa.py

```
@@ -0,0 +1,248 @@
+            bool: True if the interface is running, False otherwise.
+        """
+        if self._process is not None and self._queue_adapter is not None:
+            status = self._queue_adapter.get_status_of_job(process_id=self._process)
+            return status in ["running", "pending"]
+        else:
+            return False
+
+    def _start_process_helper(
+        self, command_lst: list[str], queue_adapter: QueueAdapter
+    ) -> int:
+        hash = hashlib.md5(str(self).encode()).hexdigest()
```

Ruff (0.12.2)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)